3DAmsterdam/Assets/Amsterdam3D/Scripts/Interface/ContextPointerMenu/ContextPointerMenu.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.EventSystems;
+using UnityEngine.UI;
+
+public class ContextPointerMenu : MonoBehaviour
+{
+	[SerializeField]
+	private RectTransform contextItemsPanel;
+
+	public static ContextPointerMenu Instance = null;
+
+	public ContextState state = ContextState.DEFAULT;
+
+	[SerializeField]
+	[Tooltip("Select buttons that should be active for specific states")]
+	private StateButtons[] buttonsAvailableOnState;
+
+	[Serializable]
+	public class StateButtons
+	{
+		public ContextState state;
+		public Button[] activeButtons;
+	}
+
+	public enum ContextState
+	{
+		DEFAULT,
+		CUSTOM_OBJECTS,
+		SELECTABLE_STATICS
+	}
+
+	private void Start()
+	{
+		if (Instance == null)
+		{
+			Instance = this;
+		}
+		contextItemsPanel.gameObject.SetActive(false);
+
+		Button[] buttons = contextItemsPanel.GetComponentsInChildren<Button>();
+		foreach (Button button in buttons)
+			button.onClick.AddListener(CloseContextMenu);
+	}
+
+	void CloseContextMenu()
+	{
+		contextItemsPanel.gameObject.SetActive(false);
+		state = ContextState.DEFAULT;
+	}
+
+	public void SwitchState(ContextState newState)
+	{
+		state = newState;
+
+		var buttons = GetComponentsInChildren<Button>();
+		foreach (Button button in buttons)
+			button.interactable = false;
+
```

Comment:
En misschien in dit script ook een beetje meer uitleg? (voor als men t in de toekomst is niet snapt)